src/main/java/uk/gov/companieshouse/api/accounts/exception/handler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.companieshouse.api.accounts.exception.handler;
+
+import static uk.gov.companieshouse.api.accounts.CompanyAccountsApplication.APPLICATION_NAME_SPACE;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import org.springframework.dao.DataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.servlet.NoHandlerFoundException;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+
+/**
+ * GlobalExceptionHandler defines handlers for generic exceptions.
+ *
+ * Api Specific Errors are handled in the Controller.
+ */
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+  @ExceptionHandler(value = {NoHandlerFoundException.class})
+  @ResponseStatus(value = HttpStatus.NOT_FOUND)
+  protected void handleNoHandlerFoundException(Exception ex) {
+    logError(ex, ExceptionMessage.NO_HANDLER_FOUND_EXCEPTION);
+  }
+
+  @ExceptionHandler(value = {DataAccessException.class, IOException.class,
+      IllegalArgumentException.class, IllegalStateException.class, NullPointerException.class,
+      NoSuchAlgorithmException.class, RuntimeException.class, Exception.class})
+  @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
```

Comment:
Wouldn't all these exceptions get recognised using the super exception types?
```java
@ExceptionHandler(value = {RuntimeException.class, Exception.class})
```

Reply:
Yes, but they were there as we needed to log specific exception message, copied from Abridged. The ExceptionMessage enum is not required at all.